components/TxConfirmationDialog.qml

```diff
@@ -0,0 +1,458 @@
+            PropertyChanges { target: bottom; visible: true }
+            PropertyChanges { target: bottomMessage; visible: false }
+            PropertyChanges { target: buttons; visible: true }
+            PropertyChanges { target: backButton; visible: true; primary: false }
+            PropertyChanges { target: confirmButton; visible: true }
+            StateChangeScript { script: confirmButton.forceActiveFocus() }
+        }, State {
+            // error message being displayed, show only back button
+            name: "error";
+            when: errorText.text !==""
+            PropertyChanges { target: dialogTitle; text: "Error" }
+            PropertyChanges { target: errorText; visible: true }
+            PropertyChanges { target: txAmountText; visible: false }
+            PropertyChanges { target: txAmountBusyIndicator; visible: false }
+            PropertyChanges { target: txFiatAmountText; visible: false }
+            PropertyChanges { target: txDetails; visible: false }
+            PropertyChanges { target: bottom; visible: true }
+            PropertyChanges { target: bottomMessage; visible: false }
+            PropertyChanges { target: buttons; visible: true }
+            PropertyChanges { target: backButton; visible: true; primary: true }
+            PropertyChanges { target: confirmButton; visible: false }
+            StateChangeScript { script: backButton.forceActiveFocus() }
+        }, State {
+            // creating or sending transaction, show tx details and don't show any button
+            name: "bottomText";
+            when: errorText.text =="" && bottomText.text !==""
+            PropertyChanges { target: errorText; visible: false }
+            PropertyChanges { target: txAmountText; visible: root.transactionAmount !== "(all)" || (root.transactionAmount === "(all)" && currentWallet.isHwBacked() === true) }
+            PropertyChanges { target: txAmountBusyIndicator; visible: !txAmountText.visible }
+            PropertyChanges { target: txFiatAmountText; visible: txAmountText.visible && persistentSettings.fiatPriceEnabled && root.transactionAmount !== "(all)" }
+            PropertyChanges { target: txDetails; visible: true }
+            PropertyChanges { target: bottom; visible: true }
+            PropertyChanges { target: bottomMessage; visible: true }
+            PropertyChanges { target: buttons; visible: false }
+        }
+    ]
+
+    // same signals as Dialog has
+    signal accepted()
+    signal rejected()
+    signal closeCallback();
+
+    function open(description, amount, address, priority) {
+        if (description) {
+            root.transactionDescription = description;  
+        }
+        if (amount) {
+            root.transactionAmount = amount;  
+        }
+        if (address) {
+            root.transactionAddress = address;  
+        }
+        if (priority) {
+            root.transactionPriority = priority;  
+        }
```

Comment:
```suggestion
        root.transactionDescription = description;
        root.transactionAmount = amount; 
        root.transactionAddress = address;  
        root.transactionPriority = priority;  
```

Reply:
in main.qml, the first time the txConfirmationPopup is opened, we send values:
`txConfirmationPopup.open(description, Utils.removeTrailingZeros(amount), address, priority)`

the second time it is opened (when the transaction is confirmed), we don't send values:
`txConfirmationPopup.open()`

So these lines are necessary, otherwise the txConfirmationPopup receive clean values when it is opened for the second time. With the transactionDescription missing, the main.qml can't do  `currentWallet.setUserNote(txid, txConfirmationPopup.transactionDescription)` and it breaks:
`W qrc:/main.qml:945: TypeError: Cannot read property 'length' of undefined`
![image](https://user-images.githubusercontent.com/45968869/84598578-0ada4880-ae6c-11ea-9f4e-0086fdc9392b.png)

Reply:
Is this when using a hardware wallet? Didn’t test that yet

Reply:
Normal wallet.

Reply:
I would recommend to do it like this. Saves lines of code and is a bit clearer :)

```diff
diff --git a/components/TxConfirmationDialog.qml b/components/TxConfirmationDialog.qml
index d1d975c8..1dc09e7d 100644
--- a/components/TxConfirmationDialog.qml
+++ b/components/TxConfirmationDialog.qml
@@ -112,20 +112,7 @@ Rectangle {
     signal rejected()
     signal closeCallback();
 
-    function open(description, amount, address, priority) {
-        if (description) {
-            root.transactionDescription = description;  
-        }
-        if (amount) {
-            root.transactionAmount = amount;  
-        }
-        if (address) {
-            root.transactionAddress = address;  
-        }
-        if (priority) {
-            root.transactionPriority = priority;  
-        }
-
+    function open() {
         root.visible = true;
         
         //clean previous error message
diff --git a/main.qml b/main.qml
index f4aa9e8c..630654af 100644
--- a/main.qml
+++ b/main.qml
@@ -841,7 +841,11 @@ ApplicationWindow {
         txConfirmationPopup.transactionDescription = description;
         txConfirmationPopup.bottomTextAnimation.running = false
         txConfirmationPopup.bottomText.text  = qsTr("Creating transaction...") + translationManager.emptyString;
-        txConfirmationPopup.open(description, Utils.removeTrailingZeros(amount), address, priority)
+        txConfirmationPopup.transactionDescription = description;
+        txConfirmationPopup.transactionAmount = Utils.removeTrailingZeros(amount); 
+        txConfirmationPopup.transactionAddress = address;  
+        txConfirmationPopup.transactionPriority = priority;  
+        txConfirmationPopup.open()
 
         // validate amount;
         if (amount !== "(all)") {
```